app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/jyou/TeclastH30Support.java

```diff
@@ -0,0 +1,484 @@
+                        cmd,
+                        alarms.get(i).isEnabled() ? cal.get(Calendar.HOUR_OF_DAY) : -1,
+                        alarms.get(i).isEnabled() ? cal.get(Calendar.MINUTE) : -1
+                ));
+            }
+            performConnected(builder.getTransaction());
+            GB.toast(getContext(), "Alarm settings applied - do note that the current device does not support day specification", Toast.LENGTH_LONG, GB.INFO);
+        } catch(IOException e) {
+            LOG.warn(e.getMessage());
+        }
+    }
+
+    @Override
+    public void onSetTime() {
+        try {
+            TransactionBuilder builder = performInitialized("SetTime");
+            syncDateAndTime(builder);
+            performConnected(builder.getTransaction());
+        } catch(IOException e) {
+            LOG.warn(e.getMessage());
+        }
+    }
+
+    @Override
+    public void onSetCallState(CallSpec callSpec) {
+        switch (callSpec.command) {
+            case CallSpec.CALL_INCOMING:
+                showNotification(JYouConstants.ICON_CALL, callSpec.name, callSpec.number);
+                break;
+        }
+    }
+
+    @Override
+    public void onSetCannedMessages(CannedMessagesSpec cannedMessagesSpec) {
+
+    }
+
+    @Override
+    public void onSetMusicState(MusicStateSpec stateSpec) {
+
+    }
+
+    @Override
+    public void onSetMusicInfo(MusicSpec musicSpec) {
+
+    }
+
+    @Override
+    public void onEnableRealtimeSteps(boolean enable) {
+        onEnableRealtimeHeartRateMeasurement(enable);
+    }
+
+    @Override
+    public void onInstallApp(Uri uri) {
+
+    }
+
+    @Override
+    public void onAppInfoReq() {
+
+    }
+
+    @Override
+    public void onAppStart(UUID uuid, boolean start) {
+
+    }
+
+    @Override
+    public void onAppDelete(UUID uuid) {
+
+    }
+
+    @Override
+    public void onAppConfiguration(UUID appUuid, String config) {
+
+    }
+
+    @Override
+    public void onAppReorder(UUID[] uuids) {
+
+    }
+
+    @Override
+    public void onFetchActivityData() {
+
+    }
+
+    @Override
+    public void onReboot() {
+        try {
+            TransactionBuilder builder = performInitialized("Reboot");
+            builder.write(ctrlCharacteristic, commandWithChecksum(
+                    JYouConstants.CMD_ACTION_REBOOT_DEVICE, 0, 0
+            ));
+            performConnected(builder.getTransaction());
+        } catch(Exception e) {
+            LOG.warn(e.getMessage());
+        }
+    }
+
+    @Override
+    public void onHeartRateTest() {
+        try {
+            TransactionBuilder builder = performInitialized("HeartRateTest");
+            builder.write(ctrlCharacteristic, commandWithChecksum(
+                    JYouConstants.CMD_ACTION_HEARTRATE_SWITCH, 0, 1
+            ));
+            performConnected(builder.getTransaction());
+        } catch(Exception e) {
+            LOG.warn(e.getMessage());
```

Comment:
When you get an exception, please pass the exception itself to warn() or error(), like this:
`LOG.warn("some message", e);`
That way we will get the actual stacktrace in the log, not just the message.

Reply:
Gotcha.